deps/github.com/arangodb/arangosync/client/client_cache.go

```diff
@@ -113,11 +116,13 @@ func NewAuthentication(tlsAuth TLSAuthentication, jwtSecret string) Authenticati
 type Authentication struct {
 	TLSAuthentication
 	JWTSecret string
+	Username  string
+	Password  string
 }
 
 // String returns a string used to unique identify the authentication settings.
 func (a Authentication) String() string {
-	return a.TLSAuthentication.String() + ":" + a.JWTSecret
+	return a.TLSAuthentication.String() + ":" + a.JWTSecret + ":" + a.Username + ":" + a.Password
```

Comment:
I do not know where String() is used.  But I wonder if making the password available via this function is required, and if not required, is it a bad thing.

Reply:
It is used to generate a key that matches the exact settings. So with exactly the same key you are allowed to share the same client object.